### PR TITLE
fix: max tick check on v3 sdk

### PR DESCRIPTION
**Pull request by Chef-Yogi**
---

## PR-Codex overview
This PR updates the `useV3DerivedInfo` hook in the V3 app with new imports, type changes, and adjustments to tick space limits.

### Detailed summary
- Added imports for `BIG_INT_ZERO` and `Bound`
- Updated type imports for `PoolState`
- Adjusted tick space limits for lower and upper bounds
- Modified tick calculation logic for fee tiers

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/v3-sdk/src/utils/tickMath.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export abstract class TickMath {
    * @param sqrtRatioX96 the sqrt ratio as a Q64.96 for which to compute the tick
    */
   public static getTickAtSqrtRatio(sqrtRatioX96: bigint): number {
-    invariant(sqrtRatioX96 >= TickMath.MIN_SQRT_RATIO && sqrtRatioX96 < TickMath.MAX_SQRT_RATIO, 'SQRT_RATIO')
+    invariant(sqrtRatioX96 >= TickMath.MIN_SQRT_RATIO && sqrtRatioX96 <= TickMath.MAX_SQRT_RATIO, 'SQRT_RATIO')
```
Comment (chefjackson) [inline packages/v3-sdk/src/utils/tickMath.ts:83]: I think we should update the price limit instead. When we set price range from 0 -> max, the sqrtRatioLimit should be [TickMath.MIN_SQRT_RATIO + 1n, TickMath.MAX_SQRT_RATIO - 1n]

**Review by chefjackson — CHANGES_REQUESTED** [review]

File: packages/v3-sdk/src/utils/tickMath.ts (hunk 1 above)
Comment (Chef-Yogi) [reply packages/v3-sdk/src/utils/tickMath.ts]: Done

Comment (chef-jojo) [reply packages/v3-sdk/src/utils/tickMath.ts]: Should only adjust when meet the max or min

File: apps/web/src/hooks/v3/useV3DerivedInfo.ts [hunk 2]
```diff
@@ -176,6 +176,8 @@ export default function useV3DerivedInfo(
           : (invertPrice && typeof rightRangeTypedValue === 'boolean') ||
             (!invertPrice && typeof leftRangeTypedValue === 'boolean')
           ? tickSpaceLimits[Bound.LOWER]
+            ? tickSpaceLimits[Bound.LOWER] + 1
```
Comment (chefjackson) [inline apps/web/src/hooks/v3/useV3DerivedInfo.ts:179]: for lower tick should be fine I think. Let's remove plus one here

File: apps/web/src/hooks/v3/useV3DerivedInfo.ts [hunk 3]
```diff
@@ -185,6 +187,8 @@ export default function useV3DerivedInfo(
           : (!invertPrice && typeof rightRangeTypedValue === 'boolean') ||
             (invertPrice && typeof leftRangeTypedValue === 'boolean')
           ? tickSpaceLimits[Bound.UPPER]
+            ? tickSpaceLimits[Bound.UPPER] - 1
```
Comment (chefjackson) [inline apps/web/src/hooks/v3/useV3DerivedInfo.ts:190]: We need to consider the tick spacing. For example if tick spacing is 10, then the tickSpaceLimit here won't be 887272, but 887270. In this case there's no need to minus 1

Comment (Chef-Yogi) [reply apps/web/src/hooks/v3/useV3DerivedInfo.ts]: Done

File: apps/web/src/hooks/v3/useV3DerivedInfo.ts (hunk 2 above)
Comment (Chef-Yogi) [reply apps/web/src/hooks/v3/useV3DerivedInfo.ts]: Done

**Review by chefjackson — APPROVED** [review]

**Pull request merged**